repository/src/main/java/org/pentaho/platform/security/userroledao/jackrabbit/AbstractJcrBackedUserRoleDao.java

```diff
@@ -444,9 +444,11 @@ public void deleteUser( Session session, final IPentahoUser user ) throws NotFou
         while ( currentGroups.hasNext() ) {
           currentGroups.next().removeMember( jackrabbitUser );
         }
+        getUserCache().remove( jackrabbitUser.getID() );
         purgeUserFromCache( user.getUsername() );
         // [BISERVER-9215]
         jackrabbitUser.remove();
+        session.save();
```

Comment:
Maybe it was just the save call that was necessary?

Reply:
The save call is definitively required. But I think the other change is required as well. See my comment on it.